sample-apps/ruby-rails-sample-app/app/controllers/application_controller.rb

```diff
@@ -0,0 +1,78 @@
+require 'net/http'
+
+def get_xray_trace_id(otel_trace_id_hex)
+    xray_trace_id = "1-#{otel_trace_id_hex[0..7]}-#{otel_trace_id_hex[8..otel_trace_id_hex.length]}"
+    { traceId: xray_trace_id }
+end
+
+class ApplicationController < ActionController::Base
+    def aws_sdk_call
+        @@tracer.in_span("get-aws-s3-buckets") do |span|
+            s3 = Aws::S3::Client.new
+            s3.list_buckets
+        end
+        render json: get_xray_trace_id(OpenTelemetry::Trace.current_span.context.hex_trace_id)
+    end
+    
+    def outgoing_http_call
+
+        @@tracer.in_span("outgoing-http-call") do |span|
+            uri = URI.parse("https://aws.amazon.com/")
+            http = Net::HTTP.new(uri.host, uri.port)
+            http.use_ssl = true
+
+            request = Net::HTTP::Get.new(uri.request_uri)
+            res = http.request(request)
+            span.set_attribute("signal", "trace")
+            span.set_attribute("language", "ruby")
+            span.set_attribute("statusCode", res.code)
+        end
+
+        render json: get_xray_trace_id(OpenTelemetry::Trace.current_span.context.hex_trace_id)
+    end
+
+    def outgoing_sampleapp
+
+        @@tracer.in_span("invoke-sample-apps") do |span|
```

Comment:
consistency: what is the reason for the name of the spans to be different/not consistent?

outgoing_http_call -> outgoing-http-call
outgoing_sampleapp-> invoke-sample-apps
leaf-request??
invoke-sampleapp??

Unrelated question: what is the reason for nesting leaf-request and invoke-sampleapp inside invoke-sample-apps? Is there anything special about this? If no, can we remove invoke-sample-apps?

Reply:
The span names should be the same across sample apps from other languages so their string value will use hyphens. The function names will depend on the language so here they are snake case.

The reason for nesting is:
invoke-sample-apps is a parent span which can have children in 2 cases. Either sampleapp ports are configured so it can have N amount of 'invoke-sampleapp' span children or no sampleapp ports are configured so it will just have a 'leaf-request'